pkg/controllers/management/nodetemplate/nodetemplate_handler.go

```diff
@@ -0,0 +1,294 @@
+package nodetemplate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/controllers/management/rbac"
+	"github.com/rancher/rancher/pkg/namespace"
+	v1 "github.com/rancher/types/apis/core/v1"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+const (
+	ownerBindingsAnno = "ownerBindingsCreated"
+)
+
+type nodeTemplateController struct {
+	roleLister v3.GlobalRoleLister
+	roleClient v3.GlobalRoleInterface
+	rbLister   v3.GlobalRoleBindingLister
+	rbClient   v3.GlobalRoleBindingInterface
+	ntClient   v3.NodeTemplateInterface
+	ntLister   v3.NodeTemplateLister
+	npLister   v3.NodePoolLister
+	npClient   v3.NodePoolInterface
+	nsLister   v1.NamespaceLister
+	nsClient   v1.NamespaceInterface
+	mgmtCtx    *config.ManagementContext
+}
+
+func Register(ctx context.Context, mgmt *config.ManagementContext) {
+	nt := nodeTemplateController{
+		roleLister: mgmt.Management.GlobalRoles("").Controller().Lister(),
+		roleClient: mgmt.Management.GlobalRoles(""),
+		rbLister:   mgmt.Management.GlobalRoleBindings("").Controller().Lister(),
+		rbClient:   mgmt.Management.GlobalRoleBindings(""),
+		ntClient:   mgmt.Management.NodeTemplates(""),
+		ntLister:   mgmt.Management.NodeTemplates("").Controller().Lister(),
+		npLister:   mgmt.Management.NodePools("").Controller().Lister(),
+		npClient:   mgmt.Management.NodePools(""),
+		nsLister:   mgmt.Core.Namespaces("").Controller().Lister(),
+		nsClient:   mgmt.Core.Namespaces(""),
+		mgmtCtx:    mgmt,
+	}
+
+	mgmt.Management.NodeTemplates("").Controller().AddHandler(ctx, "nt-grb-handler", nt.sync)
+}
+
+func (nt *nodeTemplateController) sync(key string, nodeTemplate *v3.NodeTemplate) (runtime.Object, error) {
+	if nodeTemplate == nil || nodeTemplate.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	metaAccessor, err := meta.Accessor(nodeTemplate)
+	if err != nil {
+		return nodeTemplate, err
+	}
+
+	creatorID, ok := metaAccessor.GetAnnotations()[rbac.CreatorIDAnn]
+	if !ok {
+		return nodeTemplate, fmt.Errorf("clusterTemplate %v has no creatorId annotation", metaAccessor.GetName())
+	}
+
+	// Parts of the node template in k8s are dynamic, such as an azureConfig. If the dynamic client is not use then the
+	// dynamic fields, like azureConfig, are stripped when marshalling the data into the NodeTemplate struct. If changes
+	// are made, and that struct is committed to k8s, then the dynamic field is effectively removed.
+	restConfig := nt.mgmtCtx.RESTConfig
```

Comment:
Can this be created and saved on the struct above? Then it won't need to be created on the fly and passed into the handlers